package.json

```diff
@@ -52,10 +52,13 @@
     "start": "yarn compile --watch",
     "generate-component": "yarn hygen component new",
     "prepare": "",
-    "prepublish": "in-publish && run-p test build || not-in-publish",
+    "prepublish": "in-publish && run-p -l test build || not-in-publish",
     "predeploy": "cd example && yarn install && yarn build",
-    "postpublish": "git tag v$npm_package_version && git push --tags && echo \"Successfully released version $npm_package_version!\"",
-    "release": "yarn publish --no-git-tag-version",
+    "postpublish": "git push --tags && echo \"Successfully released version $npm_package_version!\"",
+    "preversion": "run-p -l test build",
+    "version": "yarn update-changelog && git add CHANGELOG.md",
+    "new-version": "yarn version --no-git-tag-version",
+    "postnew-version": "git add . && git commit -m \"v$npm_package_version\" -n",
```

Comment:
The workflow is a little uneven in terms of what git commands are automatic (`git add . && git commit ...`) vs. manual (`git checkout master && git fetch -p && git reset ...`)

I'd probably err on the side of asking the user to do the `git` commands manually, at least at first; but it could go either way.

But I won't be a stickler about it.

Reply:
Yeah, it's kinda trickier than I would have hoped. The main problem is that we do not allow anyone to push directly to master unless they are a champion (which makes sense).

The assumption of `yarn version` and other patterns is that this ability is present. It also assumes that the `tag` is on the final commit for the version. In our case, this can also be a squashed commit from a pull request, not the original commit itself. That's why there is a _slightly_ different flow for non-champions and champions and thus the usefulness of `new-version`.